src/stumpy_core/canvas.cr

```diff
@@ -34,5 +34,36 @@ module StumpyCore
         @height == other.height &&
         @pixels == other.pixels
     end
+    
+    def paste(canvas, x, y)
+      #StumpyPNG does not handle going out of bound well so we have to be very careful with the paste
+
+      #detect left side first
+      min_x = (x < 0 ? x.abs : 0)
+      max_x = canvas.width - (x + canvas.width >= width ? x + canvas.width - width)
+      min_y = (y < 0 ? y.abs : 0)
+      max_y = canvas.height - (y + canvas.height >= height ? y + canvas.height - height)
+
+      (min_x...max_x).each do |cx|
+        (min_y...max_y).each do |cy|
+          if canvas[cx, cy].a == UInt16::MAX
+            self[x + cx, y + cy] = canvas[cx, cy]
+          elsif canvas[cx, cy].a == UInt16::MIN
+            #dont do anything, it's invisble
+          else
+            b_color = self[x + cx, y + cy]
+            t_color = canvas[cx, cy]
+
+        
+            t_a = (t_color.a.to_f/UInt16::MAX)
+            n_r = (t_color.r * t_a) + (b_color.r * (1.0 - t_a))
+            n_g = (t_color.g * t_a) + (b_color.g * (1.0 - t_a))
+            n_b = (t_color.b * t_a) + (b_color.b * (1.0 - t_a))
+
+            self[x + cx, y + cy] = RGBA.new(n_r.to_u16, n_g.to_u16, n_b.to_u16, UInt16::MAX)
```

Comment:
Is it safe to assume, that both alpha values together are = `UInt16::MAX`?

Combining images with different alpha values (!= 100%) seems to be a little bit tricky,
the [Wikipedia article on Alpha Compositing](https://en.wikipedia.org/wiki/Alpha_compositing)
has a some equations for this topic (A over B), maybe they come in helpful.